### PR TITLE
fix(highlight-flags): make ^/$ work properly

### LINKED ISSUE #8096: md-highlight-flags: ^ and $ not working properly
Using `md-autocomplete` with custom template where we have multiple span tags with `md-highlight-text` and `md-highlight-flags` the `^` and `$` flags don't work.

[Pen here](http://codepen.io/anon/pen/pyVXLP)

Reproduce:
- Type 'd'

Only the first `D` should be red, but in fact, two more gets highlighted as well.
I can try and put together a PR if needed.

### PR DESCRIPTION
**Pull request by ivoviz**
Fixes #8096

### CODE REVIEW

**Comment by DevVersion** [comment]
This looks good to me. Thanks.

I mostly use `!= -1` to check if the string is containing the query, but using `>= 0` works also.

**Comment by ThomasBurleson** [comment]
@EladBezalel - please review also.

File: src/components/autocomplete/js/highlightController.js [hunk 1]
```diff
@@ -32,10 +32,9 @@ function MdHighlightCtrl ($scope, $element, $attrs) {
   }
 
   function getRegExp (text, flags) {
-    var str = '';
-    if (flags.indexOf('^') >= 1) str += '^';
-    str += text;
-    if (flags.indexOf('$') >= 1) str += '$';
-    return new RegExp(sanitize(str), flags.replace(/[\$\^]/g, ''));
+    var b = '', e = '';
```
Comment (EladBezalel) [inline src/components/autocomplete/js/highlightController.js]: Please change to meaningful variable names

Comment (DevVersion) [inline src/components/autocomplete/js/highlightController.js]: What about `startFlag` and `endFlag`?

Comment (ivoviz) [inline src/components/autocomplete/js/highlightController.js]: Done.

**Pull request closed without merging**